docs/proposals/20230117-cluster-level-identity.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Cluster Level Identity(aka Multitenancy support)
+authors:
+- "@shyamradhakrishnan"
+- "@joekr"
+  reviewers:
+  creation-date: 2023-01-18
+  last-updated: 2023-01-18
+  status: implementable
+  see-also:
+- https://github.com/oracle/cluster-api-provider-oci/issues/203
+  replaces: []
+---
+
+## Summary
+
+Cluster API Provider for OCI(CAPOCI) supports a single OCI User Principal currently. This
+User Principal is [defined during installation][user-principal] and will be refreshed when the CAPOCI pod is 
+restarted.
+Some of the customer scenarios requires support for OCI User Principal defined per cluster. This will help
+customers create workload clusters using different OCI User Principals such that a single management cluster can 
+support a multitenanted architecture.
+
+## Goals
+
+1. To enable OCICluster resources reconciliation use a cluster specific OCI user principal.
+2. To maintain backwards compatibility and cause no impact for users who don't intend to make use of
+   this capability
```

Comment:
Possible non-goals:
- solve authentication across multiple tenancy for a single cluster. (e.g. control-plane in Tenant A, workload nodes in Tenant B)

Only reason I bring that up is I think someone in the past has asked for that ability

Reply:
good callout, will add